docs/docs/concepts/blueprints.md

```diff
@@ -1,61 +1,153 @@
+| contained     | boolean  | Describes the conceptual belonging of one entity to another. In other words, if an entity must always be considered in the context of it's "parent" (true), or the entity has meaning on it's own (false). |
+| dimensions    | string   | Degree of dimensions. Format; A list of integers, each integer defines the size of a dimension. "*"(wildcard) has the special meaning of "unfixed size                                                     |
+| optional      | boolean  | If the attribute is optional or required                                                                                                                                                                   |
+| default       | any json | Default value for the attribute. Same type as the attribute itself                                                                                                                                         |
+| enumType      | string   | A reference to an Enum entity. Value should hold the chosen enum value in the entity                                                                                                                       |
+
+## References 
+
+It is possible to reference other blueprint from the `type`, `extends`, `attributeType`, and `_blueprintPath_` attributes.
+
+### Syntax
+
+There are several ways to describe a reference to other blueprints. 
 
-It says that a Car can have an attribute `name` and `description`.
+All of these references are understood and resolved during import. 
 
-So an instance of a Car can look like.
+*By URI* - Full absolute path prefixed with protocol
+
+```
+dmss://datasource/package/entity
+dmss://datasource/package/subfolder/entity
+```
+
+*By Alias* - Require dependencies to be defined somewhere in the source tree using [_meta_](./../concepts/meta.md) information.
+
+```
+ALIAS:packge/entity
+ALIAS:entity
+```
+
+*By data source* - Relative from the destination data source.
+
+```
+/package/entity
+/package/subfolder/entity
+```
+
+*By package* - Relative from the source package
+
+```
+entity
+subfolder/entity
+```
+
+*By dotted* - Relative from the file (UNIX directory traversal)
+
+```
+./../entity
+../subfolder/entity
+../../subfolder/entity
+```
+
+## Entity
+
+An entity populates the attributes of the blueprint with real values.
+
+An entity of the `Car` blueprint can look like this. 
 
 ```json
 {
+   "type": "dmss://Car",
    "name": "Volvo",
    "description": "This is an typically car type used in Norway"
 }
 ```
 
-#### Blueprint attributes
-
-Any blueprint can be understood by simply looking at its attributes. Each attribute has several options, some are required and other are optional.
+By looking at the `type` keyword, we can see that it's of type `Car`.
```

Comment:
I think this should be moved up, so that it is positioned right after What is a Blueprint?, since it helps understanding what a blueprint is.

Reply:
Good idea, I moved it.